src/components/application_manager/src/help_prompt_manager_impl.cc

```diff
@@ -0,0 +1,426 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include "application_manager/application.h"
+#include "application_manager/help_prompt_manager_impl.h"
+#include "application_manager/message_helper.h"
+#include "application_manager/smart_object_keys.h"
+#include "smart_objects/smart_object.h"
+#include "application_manager/application_manager.h"
+#include "application_manager/commands/command_impl.h"
+#include "utils/logger.h"
+#include "utils/timer_task_impl.h"
+#include "utils/make_shared.h"
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "HelpPromptManagerImpl")
+
+namespace {
+const unsigned kLimit_command = 30;
+const uint32_t kBuffering_timeout = 10000;
+}
+
+namespace application_manager {
+
+HelpPromptManagerImpl::HelpPromptManagerImpl(Application& app,
+                                             ApplicationManager& app_manager)
+    : internal_timer_("HelpPromtManagerTimer",
+                      new ::timer::TimerTaskImpl<HelpPromptManagerImpl>(
+                          this, &HelpPromptManagerImpl::OnTimerExpired))
+    , app_(app)
+    , app_manager_(app_manager)
+    , sending_type_(SendingType::kNoneSend)
+    , count_requests_commands_(0)
+    , is_tts_send_(false)
+    , is_ui_send_(false) {}
+
+HelpPromptManagerImpl::~HelpPromptManagerImpl() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  StopTimer();
+}
+
+void HelpPromptManagerImpl::OnVrCommandAdded(
+    uint32_t cmd_id, const smart_objects::SmartObject& command) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (SendingType::kNoneSend == sending_type_) {
+    return;
```

Comment:
@SNiukalov please add log messages for all returns inside if's

Reply:
Thanks, fixed in https://github.com/smartdevicelink/sdl_core/pull/2186/commits/766c28211da10784fcd3f311a6e1ee24019fd007